working/1426-extension-types/feature-specification.md

```diff
@@ -888,15 +895,28 @@ extension type, e.g., `myList is List<nat>`.*
 We could use the following mechanism to enable casts to a protected
 extension type to succeed at run time:
 
-Assume that `E` is a protected extension type that declares a `bool get
-verifyThis` getter.
+Assume that `E` is a protected extension type that declares a
+`bool get verifyThis` getter.
 
 If such a getter exists, then the execution of a type cast `c` of the form
 `o as E` proceeds as follows: First, a cast `o as T` is executed, where `T`
 is the instantiated on-type corresponding to `E`. If this cast succeeds
-then `o.verifyThis` is evaluated to an object `o1`. If `o1` is the true
-object then `c` completes normally and yields `o`; otherwise `c` encounters
-a dynamic type error.
+then `this` is bound to `o` and `verifyThis` is evaluated to an object
+`o1`. If `o1` is the true object then `c` completes normally and yields
+`o`; otherwise `c` encounters a dynamic type error.
+
+*This implies that it is possible to execute `verifyThis` as declared in
+`E` in a situation where `this` is bound to an object that has not
+necessarily been returned by a constructor in `E`, and which isn't
+guaranteed to make `verifyThis` return true. In other words, this seems to
+be a violation of the discipline associated with protected extension types,
+because that object "isn't worthy of being `this` for the execution of any
+code in `E`". However, it seems more natural to access the "candidate
+`this`" using `this` than it would be if we were to use a different
+declaration to perform the verification (say, a static function in `E`). 
+Also, it is obvious that the `this` in the body of `verifyThis` may not
+satisfy the requirements, and we trust developers to write the
+implementation of `verifyThis` with that fact in mind.*
```

Comment:
Consider calling it `operator is()` instead. :grin:

Reply:
Could do that! (But I'm working on extension methods now, will return to this one soon ;-)